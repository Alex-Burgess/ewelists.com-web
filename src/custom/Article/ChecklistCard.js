import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/icons
import AssignmentTurnedIn from "@material-ui/icons/ListAlt";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";

import styles from "assets/jss/custom/components/article/checklistCardStyle.js";
const useStyles = makeStyles(styles);

export default function ChecklistCard(props) {
  const { items, packing } = props;
  const classes = useStyles();

  const title = () => {
    if (packing) {
        return "Items for your packing list"
    } else {
      return "Items for your list"
    }
  }

  return (
    <Card className={classes.checklistCard}>
      <CardBody color>
        <h5 className={classes.cardCategorySocialWhite}>
          <AssignmentTurnedIn />
          {title()}
        </h5>
        <h5 className={classes.cardTitleWhite}>
          <GridContainer >
            <GridItem xs={12} sm={6} md={6}>
              <ul className={classes.ulCustom}>
                {items.map((item, i) =>
                  i % 2
                  ? null
                  : <li className={classes.listSpacing} key={i}>{item}</li>
                )}
              </ul>
            </GridItem>
            <GridItem xs={12} sm={6} md={6}>
              <ul className={classes.ulCustom}>
                {items.map((item, i) =>
                  i % 2
                  ? <li className={classes.listSpacing} key={i}>{item}</li>
                  : null
                )}
              </ul>
            </GridItem>
          </GridContainer>
        </h5>
      </CardBody>
    </Card>
  );
}

ChecklistCard.propTypes = {
  items: PropTypes.array,
  packing: PropTypes.bool
};
