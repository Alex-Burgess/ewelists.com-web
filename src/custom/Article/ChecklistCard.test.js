import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChecklistCard from './ChecklistCard';

const items = ['Cool comfy clothing', 'Change of clothes', 'Phone', 'Camera', 'Charger', 'Swimwear for the pool'];

test('Renders renderer component.', () => {

  const tree = renderer
    .create(
      <Router>
        <ChecklistCard
          items={items}
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
