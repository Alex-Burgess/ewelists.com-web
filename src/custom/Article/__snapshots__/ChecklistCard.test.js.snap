// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders renderer component. 1`] = `
<div
  className="makeStyles-card-7 makeStyles-checklistCard-6"
>
  <div
    className="makeStyles-cardBody-22 makeStyles-cardBodyColor-28"
  >
    <h5
      className="makeStyles-cardCategorySocialWhite-4"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m1.1-2H3.9c-.5 0-.9.4-.9.9v16.2c0 .4.4.9.9.9h16.2c.4 0 .9-.5.9-.9V3.9c0-.5-.5-.9-.9-.9zM11 7h6v2h-6V7zm0 4h6v2h-6v-2zm0 4h6v2h-6zM7 7h2v2H7zm0 4h2v2H7zm0 4h2v2H7z"
        />
      </svg>
      Items for your packing list
    </h5>
    <h5
      className="makeStyles-cardTitleWhite-5"
    >
      <div
        className="MuiGrid-root makeStyles-grid-38  MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-grid-142  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6"
        >
          <ul
            className="makeStyles-ulCustom-1"
          >
            <li
              className="makeStyles-listSpacing-2"
            >
              Cool comfy clothing
            </li>
            <li
              className="makeStyles-listSpacing-2"
            >
              Phone
            </li>
            <li
              className="makeStyles-listSpacing-2"
            >
              Charger
            </li>
          </ul>
        </div>
        <div
          className="MuiGrid-root makeStyles-grid-142  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-6"
        >
          <ul
            className="makeStyles-ulCustom-1"
          >
            <li
              className="makeStyles-listSpacing-2"
            >
              Change of clothes
            </li>
            <li
              className="makeStyles-listSpacing-2"
            >
              Camera
            </li>
            <li
              className="makeStyles-listSpacing-2"
            >
              Swimwear for the pool
            </li>
          </ul>
        </div>
      </div>
    </h5>
  </div>
</div>
`;
